File: app/src/main/java/com/uici/repertoire/ContactManager.java
package com.uici.repertoire;

import android.content.Context;
import android.content.SharedPreferences;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class ContactManager {
    private SharedPreferences sharedPreferences;
    private Gson gson;

    public ContactManager(Context context) {
        this.sharedPreferences = context.getSharedPreferences("contacts_prefs", Context.MODE_PRIVATE);
        this.gson = new Gson();
    }

    public void saveContacts(List<Contact> contacts) {
        String json = gson.toJson(contacts);
        sharedPreferences.edit().putString("contacts", json).apply();
    }

    public List<Contact> getContacts() {
        String json = sharedPreferences.getString("contacts", null);
        if (json != null) {
            Type type = new TypeToken<ArrayList<Contact>>() {}.getType();
            return gson.fromJson(json, type);
        } else {
            return new ArrayList<>();
        }
    }

    public void addContact(Contact contact) {
        List<Contact> contacts = getContacts();
        contacts.add(contact);
        saveContacts(contacts);
    }

    public void deleteContact(String contactId) {
        List<Contact> contacts = getContacts();
        contacts.removeIf(contact -> contact.getId().equals(contactId));
        saveContacts(contacts);
    }
}

File: app/src/main/java/com/uici/repertoire/ContactManager.kt
package com.uici.repertoire

import android.content.Context
import android.content.SharedPreferences
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class ContactManager(context: Context) {
    private val sharedPreferences: SharedPreferences =
        context.getSharedPreferences("contacts_prefs", Context.MODE_PRIVATE)
    private val gson = Gson()

    fun saveContacts(contacts: List<Contact>) {
        val json = gson.toJson(contacts)
        sharedPreferences.edit().putString("contacts", json).apply()
    }

    fun getContacts(): MutableList<Contact> {
        val json = sharedPreferences.getString("contacts", null)
        return if (json != null) {
            val type = object : TypeToken<MutableList<Contact>>() {}.type
            gson.fromJson(json, type)
        } else {
            mutableListOf()
        }
    }

    fun addContact(contact: Contact) {
        val contacts = getContacts()
        contacts.add(contact)
        saveContacts(contacts)
    }

    fun deleteContact(contactId: String) {
        val contacts = getContacts()
        contacts.removeAll { it.id == contactId }
        saveContacts(contacts)
    }
}
